src/components/Api.js

```diff
@@ -0,0 +1,127 @@
+export class Api {
+  constructor({ baseUrl, headers, contentType }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    this._contentType = contentType;
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(`${this._baseUrl}/cards`, {
+        method: "GET",
+        headers: {
+          authorization: this._headers,
+        },
+      });
+
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+
+      return await res.json(); // si pasa el error, devuelve la respuesta como JASON (objeto)
+    } catch (err) {
+      // y sino devuelve un error
+      console.error("I got an error:", err.message);
+      throw err;
+    }
+  }
+  async getUserInfo() {
+    const res = await fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: {
+        authorization: this._headers,
+      },
+    });
+
+    if (res.ok) {
```

Comment:
Needs correction: The code for processing server response gets duplicated. You should declare a method and reuse it

```js
_processResponse = (res) => { 
      if (res.ok) { 
        return res.json(); 
      } 
      return Promise.reject(`Error: ${res.status}`); 
}

getInitialCards() { 
  ...
  return this._processResponse(res);
}
```

Reply:
Could I get some advice on how to resolve this correction

Reply:
You should add the code from the example to all the api methods:

```js
getInitialCards() { 
  const res = await ..;

  // this converts the server response to json
  return this._processResponse(res);
}

// the same for other methods
```

Reply:
oh perfect I understood!! its a way to handle the respond with a callback function and transform it in json

Reply:
ill aplly it to the other calls, thank you so much